### PR TITLE
Apr 638 extend organisation type with category

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.RoATPService.Application.Api/Controllers/LookupDataController.cs [hunk 1]
```diff
@@ -32,14 +32,31 @@ public async Task<IActionResult> ProviderTypes()
         }
 
         [HttpGet]
-        [Route("organisationTypes")]
+        [Route("organisationTypes/{providerTypeId}")]
```
Comment (David-B-Read) [inline src/SFA.DAS.RoATPService.Application.Api/Controllers/LookupDataController.cs]: Is this a breaking change for the API clients in admin service?

Comment (David-B-Read) [reply src/SFA.DAS.RoATPService.Application.Api/Controllers/LookupDataController.cs]: https://github.com/SkillsFundingAgency/das-admin-service/blob/a1748d828e746d782d99a7ba476441c3bca0b5ab/src/SFA.DAS.AdminService.Web/Infrastructure/RoatpApiClient.cs#L56

File: src/SFA.DAS.RoATPService.Application/Handlers/GetORganisationCategoriesHandler.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿using System;
```
Comment (David-B-Read) [inline src/SFA.DAS.RoATPService.Application/Handlers/GetORganisationCategoriesHandler.cs]: Really picky, but can we fix the capitalisation of this filename?

File: src/SFA.DAS.RoATPService.Application/Handlers/GetORganisationCategoriesHandler.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        public async Task<IEnumerable<OrganisationCategory>> Handle(GetOrganisationCategoriesRequest request, CancellationToken cancellationToken)
+        {
+            if (!_providerTypeValidator.IsValidProviderTypeId(request.ProviderTypeId))
+            {
+                string invalidProviderTypeError = $@"Invalid Provider Type Id [{request.ProviderTypeId}]";
+                _logger.LogInformation(invalidProviderTypeError);
+                throw new BadRequestException(invalidProviderTypeError);
+            }
+
+            _logger.LogInformation($@"Handling Organisation Types lookup for Provider Type Id [{request.ProviderTypeId}]");
```
Comment (David-B-Read) [inline src/SFA.DAS.RoATPService.Application/Handlers/GetORganisationCategoriesHandler.cs]: This log message should be Handling Organisation Categories lookup, I think?

File: src/SFA.DAS.RoATPService.Application/Handlers/GetORganisationCategoriesHandler.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+            try
+            {
+                return await _repository.GetOrganisationCategories(request.ProviderTypeId);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Unable to retrieve Organisation Types", ex);
```
Comment (David-B-Read) [inline src/SFA.DAS.RoATPService.Application/Handlers/GetORganisationCategoriesHandler.cs]: Same here

File: src/SFA.DAS.RoATPService.Application/Handlers/GetOrganisationTypesByCategoryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        public async Task<IEnumerable<OrganisationType>> Handle(GetOrganisationTypesByCategoryRequest request, CancellationToken cancellationToken)
+        {
+            if (!_providerTypeValidator.IsValidProviderTypeId(request.ProviderTypeId))
+            {
+                string invalidProviderTypeError = $@"Invalid Provider Type Id [{request.ProviderTypeId}]";
+                _logger.LogInformation(invalidProviderTypeError);
+                throw new BadRequestException(invalidProviderTypeError);
+            }
+
```
Comment (David-B-Read) [inline src/SFA.DAS.RoATPService.Application/Handlers/GetOrganisationTypesByCategoryHandler.cs]: Should we validate the category id too?

File: src/SFA.DAS.RoATPService.Application/Handlers/GetOrganisationTypesByCategoryHandler.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            _logger.LogInformation($@"Handling Organisation Types lookup for Provider Type Id [{request.ProviderTypeId}]");
```
Comment (David-B-Read) [inline src/SFA.DAS.RoATPService.Application/Handlers/GetOrganisationTypesByCategoryHandler.cs]: Add category id to the log message?

File: src/SFA.DAS.RoATPService.Application/Handlers/GetORganisationCategoriesHandler.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.RoATPService.Application/Handlers/GetORganisationCategoriesHandler.cs]: I have fixed this, but this doesn't show as 'outdated'

**Pull request merged**